cloud/db/organizations.ts

```diff
@@ -0,0 +1,493 @@
+  // Base Implementation
+  // ---------------------------------------------------------------------------
+
+  protected getResourceName(): string {
+    return "organization";
+  }
+
+  protected getPermissionTable(): PermissionTable<OrganizationRole> {
+    return {
+      create: ["OWNER"], // Handled separately (no org exists yet)
+      read: ["OWNER", "ADMIN", "MEMBER"],
+      update: ["OWNER", "ADMIN"],
+      delete: ["OWNER"],
+    };
+  }
+
+  // ---------------------------------------------------------------------------
+  // Role Resolution
+  // ---------------------------------------------------------------------------
+
+  /**
+   * Determines the user's role in an organization.
+   *
+   * Delegates to the memberships service since the role is stored in the
+   * organization_memberships table.
+   *
+   * @param args.organizationId - The organization to check membership in
+   * @param args.userId - The user to check
+   * @returns The user's role in the organization
+   * @throws NotFoundError - If the user is not a member (hides org existence)
+   * @throws DatabaseError - If the database query fails
+   */
+  getRole({
+    userId,
+    organizationId,
+  }: {
+    userId: string;
+    organizationId: string;
+  }): Effect.Effect<
+    OrganizationRole,
+    NotFoundError | DatabaseError,
+    DrizzleORM
+  > {
+    return this.memberships.getRole({ userId, organizationId });
+  }
+
+  // ---------------------------------------------------------------------------
+  // CRUD Operations
+  // ---------------------------------------------------------------------------
+
+  /**
+   * Creates a new organization and assigns the creator as OWNER.
+   *
+   * This is a special case that doesn't require prior authorization since
+   * no organization exists yet. The creating user automatically becomes
+   * the organization's owner.
+   *
+   * Performs an atomic transaction:
+   * 1. Insert the organization
+   * 2. Create an OWNER membership for the user
+   * 3. Create an audit log for the OWNER grant
+   *
+   * @param args.userId - The user creating the organization (becomes OWNER)
+   * @param args.data - Organization data (name)
+   * @returns The created organization with the user's role
+   * @throws AlreadyExistsError - If an organization with this name exists
+   * @throws DatabaseError - If the database operation fails
+   */
+  create({
+    userId,
+    data,
+  }: {
+    userId: string;
+    data: NewOrganization;
+  }): Effect.Effect<
+    PublicOrganizationWithMembership,
+    AlreadyExistsError | NotFoundError | PermissionDeniedError | DatabaseError,
+    DrizzleORM
+  > {
+    return Effect.gen(function* () {
+      const client = yield* DrizzleORM;
+
+      return yield* client.withTransaction(
+        Effect.gen(function* () {
+          // Insert the organization
+          const results = yield* client
+            .insert(organizations)
+            .values({ name: data.name })
```

Comment:
Should we be normalizing names at all?  
  
It seems wrong that I could create a `Mirascope` org after you already created a `mirascope` org  
  
Also if we are making names unique I assume it is bc we want to use them as slugs in urls at some point. So probably we should change the schema to have something like `slugName` (validated: all lowercase, no special characters, unique, etc) and `displayName`(more permissive)